src/main/java/ua/danit/final_project/configuration/security/JwtTokenUtil.java

```diff
@@ -0,0 +1,124 @@
+package ua.danit.final_project.configuration.security;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Clock;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.impl.DefaultClock;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Component;
+import ua.danit.final_project.entities.User;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+@Component
+public class JwtTokenUtil {
+
+  private final Logger logger = LoggerFactory.getLogger(JwtTokenUtil.class);
+
+  private Clock clock = DefaultClock.INSTANCE;
+
+  @Value("${jwt.secret}")
+  private String secret;
+
+  @Value("${jwt.expiration}")
+  private Long expiration;
+
+  public String getUsernameFromToken(String token) {
+    return getClaimFromToken(token, Claims::getSubject);
+  }
+
+  public Date getIssuedAtDateFromToken(String token) {
+    return getClaimFromToken(token, Claims::getIssuedAt);
+  }
+
+  public Date getExpirationDateFromToken(String token) {
+    return getClaimFromToken(token, Claims::getExpiration);
+  }
+
+  public <T> T getClaimFromToken(String token, Function<Claims, T> claimsResolver) {
+    final Claims claims = getAllClaimsFromToken(token);
+    return claimsResolver.apply(claims);
+  }
+
+  private Claims getAllClaimsFromToken(String token) {
+    return Jwts.parser()
+        .setSigningKey(secret)
+        .parseClaimsJws(token)
+        .getBody();
+  }
+
+  private Boolean isTokenExpired(String token) {
+    final Date expiration = getExpirationDateFromToken(token);
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Rename "expiration" which hides the field declared at line 30. ([squid:HiddenFieldCheck](https://sonarcloud.io/organizations/final-project/rules#rule_key=squid%3AHiddenFieldCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=final-project&pullRequest=173&issues=AWb1IgB1W3Y69bz5KWFt&open=AWb1IgB1W3Y69bz5KWFt)

[ISSUE_KEY]: AWb1IgB1W3Y69bz5KWFt